src/Sylius/Component/Grid/Filter/MoneyFilter.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Grid\Filter;
+
+use Sylius\Component\Grid\Data\DataSourceInterface;
+use Sylius\Component\Grid\Filtering\FilterInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class MoneyFilter implements FilterInterface
+{
+    const NAME = 'money';
+
+    const DEFAULT_SCALE = 2;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(DataSourceInterface $dataSource, $name, $data, array $options)
+    {
+        if (empty($data) || empty($data['currency'])) {
+            return;
+        }
+
+        $field = isset($options['field']) ? $options['field'] : $name;
+        $scale = isset($options['scale']) ? (int) $options['scale'] : self::DEFAULT_SCALE;
+
+        $greaterThan = $this->getDataValue($data, 'greaterThan');
+        $lessThan = $this->getDataValue($data, 'lessThan');
+
+        $expressionBuilder = $dataSource->getExpressionBuilder();
+
+        if ('' !== $greaterThan) {
+            $expressionBuilder->greaterThan($field, $this->normalizeAmount($greaterThan, $scale));
+        }
+        if ('' !== $lessThan) {
+            $expressionBuilder->lessThan($field, $this->normalizeAmount($lessThan, $scale));
+        }
+
+        $dataSource->restrict($expressionBuilder->equals($options['currency_field'], $data['currency']));
```

Comment:
If I have a single currency shop I won't need to select currency each time :) Since it's a zero-cost change and we should be as generic as possible I think we should not limit that use case :)